packages/test-utils/lib/e2e/clean-appointment.ts

```diff
@@ -24,3 +36,186 @@ export const cleanAppointment = async (appointmentId: string, env: string): Prom
     return false;
   }
 };
+
+export const cleanAppointmentGraph = async (tag: Coding, oystehr: Oystehr): Promise<boolean> => {
+  const allResources = await getAppointmentGraphByTag(oystehr, tag);
+  const [deleteRequests, persons] = generateDeleteRequestsAndPerson(allResources);
+
+  await Promise.all(persons.map((person) => patchPerson(oystehr, person, allResources)));
+
+  try {
+    console.log(`Deleting resources...`);
+    const transaction = await oystehr.fhir.transaction({ requests: [...deleteRequests] });
+    console.log(`Transaction response:`, JSON.stringify(transaction, null, 2));
+    return true;
+  } catch (e) {
+    console.log(`Error deleting resources: ${e}`, JSON.stringify(e));
+    return false;
+  }
+};
+
+// these things are important environment infra rather than ephemeral test data, so leave them alone
+const NEVER_DELETE = [
+  'Person',
+  'Location',
+  'Schedule',
+  'Organization',
+  'Practitioner',
+  'PractitionerRole',
+  'HealthcareService',
+];
+
+const generateDeleteRequestsAndPerson = (allResources: FhirResource[]): [BatchInputDeleteRequest[], Person[]] => {
+  const deleteRequests: BatchInputDeleteRequest[] = [];
+
+  const persons = allResources.filter((resourceTemp) => resourceTemp.resourceType === 'Person') as Person[];
+
+  deleteRequests.push(
+    ...allResources.flatMap((resourceTemp) => {
+      if (NEVER_DELETE.includes(resourceTemp.resourceType)) {
+        return [] as BatchInputDeleteRequest[];
+      } else {
+        return { method: 'DELETE', url: `${resourceTemp.resourceType}/${resourceTemp.id}` } as BatchInputDeleteRequest;
+      }
+    })
+  );
+
+  console.log('Resources to be deleted:');
+  deleteRequests.forEach((request) => {
+    const [resourceType, id] = request.url.slice(1).split('/');
```

Comment:
wouldn't the `slice(1)` here essentially remove the first letter of the resource type and guarantee failure?